lib/webvalve.rb

```diff
@@ -33,16 +35,42 @@ def config_paths
       @config_paths ||= Set.new
     end
 
+    def logger
+      @logger ||= build_logger
+    end
+
+    def logger=(logger)
+      @logger = logger
+    end
+
+    def env
+      @env ||= (ENV['RAILS_ENV'] || ENV['RACK_ENV'] || 'development').inquiry
+    end
+
+    def env=(env)
+      @env = env&.inquiry
+    end
+
     private
 
+    def build_logger
+      if defined?(::Rails)
+        ::Rails.logger
+      else
+        ActiveSupport::Logger.new(STDOUT).tap do |l|
+          l.formatter = ::Logger::Formatter.new
+        end
+      end
+    end
```

Comment:
Like the example above, you could conditionally define the logger at class load time. Not a big deal, but it would mean not having to memoize the decision the first time the codepath is hit.

```ruby
if defined?(::Rails)
  delegate :logger, to: ::Rails
else
  def logger
    ActiveSupport::Logger.new(STDOUT).tap do |l|
      l.formatter = ::Logger::Formatter.new
    end
  end
end
```

I think that would be pretty safe, since you're also conditionally loading the railtie at class load time. I imagine you could group them all together and only ask `defined?(::Rails)` once.